public/docs/ts/latest/guide/structural-directives.jade

```diff
@@ -0,0 +1,647 @@
+include ../../../../_includes/_util-fns
+
+:marked
+  One of the defining features of a single page application is its manipulation
+  of the DOM tree. Instead of serving a whole new page every time a user
+  navigates, whole sections of the DOM disappear and are reconstructed according 
+  to the application state. In this chapter we'll to look at how Angular
+  does this DOM manipulation, and how we can do it ourselves in our own directives.
+  
+  ### Table of contents
+  
+  - [Structural directives and element instantiation](#structural-directives-and-element-instantiation)
+  - [Presence versus visibility](#presence-versus-visibility)
+  - [Behind the hood: templates](#behind-the-hood-templates)
+  - [Making our own structural directive](#making-our-own-structural-directive)
+  - [The code so far](#the-code-so-far)
```

Comment:
I think this TOC is too fine grained and it isn't always helpful.  Why would someone want to navigate to "the code so far".

Let's revisit this.